src/plugins/dashboard/public/dashboard_container/embeddable/create/search_sessions/new_session.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Filter, TimeRange, onlyDisabledFiltersChanged } from '@kbn/es-query';
+import { combineLatest, distinctUntilChanged, Observable, skip } from 'rxjs';
+import { shouldRefreshFilterCompareOptions } from '@kbn/embeddable-plugin/public';
+import { apiPublishesSettings } from '@kbn/presentation-containers/interfaces/publishes_settings';
+import { apiPublishesUnifiedSearch } from '@kbn/presentation-publishing';
+import { areTimesEqual } from '../../../state/diffing/dashboard_diffing_utils';
+import { DashboardContainer } from '../../dashboard_container';
+
+export function newSession$(api: unknown) {
```

Comment:
Do you think it's worth making this function take an `unknown` api when we actually _do_ know the type here?

We know it's actually a Dashboard container, and we're in the `dashboard` plugin.

Reply:
Chatted offline, going to keep as unknown to make testing simpler. This allows tests to just pass in parts of dashboarContainer with complex mocking or casting.